src/utils_lua.c

```diff
@@ -274,38 +235,200 @@ int luaC_pushvalue(lua_State *L, value_t v, int ds_type) /* {{{ */
+    /* meta = {} */
+    lua_setfield(L, -2, "meta");
+    return 0;
+  }
+  for (int i = 0; i < meta_count; i++) {
+    int meta_type = meta_data_type(meta, toc[i]);
+    int exist = meta_data_exists(meta, toc[i]);
+    if (!exist) {
+      toc++;
+      continue;
+    }
+    lua_newtable(L);
+    lua_pushinteger(L, meta_type);
+    lua_setfield(L, -2, "type");
+    char *string;
+    int64_t i64value;
+    uint64_t u64value;
+    double dvalue;
+    bool bvalue;
+    switch (meta_type) {
+    case MD_TYPE_STRING:
+      meta_data_get_string(meta, toc[i], &string);
+      lua_pushstring(L, string);
+      break;
+    case MD_TYPE_SIGNED_INT:
+      meta_data_get_signed_int(meta, toc[i], &i64value);
+      lua_pushnumber(L, i64value);
+      break;
+    case MD_TYPE_UNSIGNED_INT:
+      meta_data_get_unsigned_int(meta, toc[i], &u64value);
+      lua_pushnumber(L, u64value);
+      break;
+    case MD_TYPE_DOUBLE:
+      meta_data_get_double(meta, toc[i], &dvalue);
+      lua_pushnumber(L, dvalue);
+      break;
+    case MD_TYPE_BOOLEAN:
+      meta_data_get_boolean(meta, toc[i], &bvalue);
+      lua_pushboolean(L, bvalue);
+      break;
+    default:
+      /* should not happen */
+      break;
+    }
+    if (MD_TYPE_STRING >= meta_type && meta_type <= MD_TYPE_BOOLEAN) {
+      lua_setfield(L, -2, toc[i]);
+    }
+    /* meta[N] = {} */
+    lua_rawseti(L, -2, i + 1);
+  }
+  /* meta => {} */
+  lua_setfield(L, -2, "meta");
+  return 0;
+} /* }}} int luaC_pushmetadata */
 
-  lua_pushstring(L, vl->plugin);
-  lua_setfield(L, -2, "plugin");
-  lua_pushstring(L, vl->plugin_instance);
-  lua_setfield(L, -2, "plugin_instance");
+int luaC_pushmetriclist(lua_State *L, metric_list_t metric) /* {{{ */
+{
+  /*
+    metric_list_t will be mapped to the following Lua table:
+
+    metric = {
+      [1] => {
+        label => ...,
+        value => ...,
+        time => ...,
+        interval => ...,
+        },
+      ...
+      [N] => {
+       ...
+      }
+   */
+  lua_newtable(L);
+  if (metric.num > 0) {
+    metric_t *ptr = metric.ptr;
+    for (int i = 0; i < metric.num; i++) {
+      lua_newtable(L);
+      luaC_pushlabelset(L, ptr->label);
+      lua_setfield(L, -2, "label");
+      luaC_pushvalue(L, ptr->value, ptr->family->type);
+      lua_setfield(L, -2, "value");
+      luaC_pushcdtime(L, ptr->time);
+      lua_setfield(L, -2, "time");
+      luaC_pushcdtime(L, ptr->interval);
+      lua_setfield(L, -2, "interval");
+      /* metric[N] = {} */
+      lua_rawseti(L, -2, i + 1);
+      ptr++;
+    }
+  }
+  return 0;
+} /* }}} int luaC_pushmetriclist */
 
-  lua_pushstring(L, vl->type);
+int luaC_pushmetricfamily(lua_State *L, metric_family_t const *mf) /* {{{ */
+{
+  /*
+    metric_family_t is mapped to the following Lua table:
+
+    {
+      name => ...,
+      help => ...,
+      unit => ...,
+      type => ...,
+      resource => {
+        [1] => { name => value }
+        ...,
+        [N] => { ... }
+      },
+      metric => {
```

Comment:
Maybe abbreviate this to `metric => luaC_pushmetriclist(),` to keep the comment succinct.

Reply:
Thanks. Fixed it.